Readme 2.0 (#84)
sunnyhaibin/sunnypilot: README.md

* [Highlight Features](#-highlight-features)
* [Driving Enhancements](#-driving-enhancements)
* [Branch Definitions](#-branch-definitions)
* [Recommended Branches](#-recommended-branches)
* [How-To's](#-How-Tos)
* [Donate](#-donate)
* [Pull Requests](#-Pull-Requests)
* [Special Thanks](#-special-thanks)
* [Licensing](#licensing)

---

<details><summary><h3>💭 Join our Discord</h3></summary>

 Join the official sunnypilot Discord server to stay up to date with all the latest features and be a part of shaping the future of sunnypilot!
* https://discord.gg/wRW3meAgtx
 
    ![Discord Shield](https://discordapp.com/api/guilds/880416502577266699/widget.png?style=shield)
 
</details>

<details><summary><h3>🚨 Read Before Installing</h3></summary>
 
It is recommended to read this entire page before proceeding. This will ensure that you fully understand each added feature on sunnypilot, and you are selecting the right branch for your car (and device) to have the best driving experience.

sunnypilot is compatible with **most** models of the following car manufacturers:
* Hyundai/Kia/Genesis (**HKG**)
* Honda/Acura
* Toyota/Lexus
* Subaru
* Chrysler/Jeep/RAM (**FCA**)
* General Motors (**GM**)

> * VW/Audi/Skoda

Reviewer: ```suggestion
* Volkswagen (MQB/PQ)/Audi/Skoda
```

This is a fork of [comma.ai's openpilot](https://github.com/commaai/openpilot). By installing this software, you accept all responsibility for anything that might occur while you use it. All contributors to sunnypilot are not liable. ❗<ins>**Use at your own risk.**</ins>❗
</details>

<details><summary><h3>⛔ Prohibited Safety Modifications</h3></summary>
 

> All [official sunnypilot branches](https://github.com/sunnyhaibin/sunnypilot/branches) strictly adhere to [comma.ai's safety policy](https://github.com/commaai/openpilot/blob/master/docs/SAFETY.md). Any changes to this policy will result in your fork and/or device being banned from both comma.ai and sunnypilot channels.

Reviewer: ```suggestion
All [official sunnypilot branches](https://github.com/sunnyhaibin/sunnypilot/branches) strictly adhere to [comma.ai's safety policy](https://github.com/commaai/openpilot/blob/master/docs/SAFETY.md). Any changes in the software that alter the legitimacy or enforcement of this policy may result in your fork and/or device being banned from both comma.ai and sunnypilot channels.
```
This may sound more explicit. What do you think?

Author: I do agree, how about something like:  

> Any changes to* the software that alter ~~the legitimacy or enforcement of~~ this policy may result in your fork and/or device being banned from both comma.ai and sunnypilot channels.

Less is more seen as we are referring to the policy in the sentence before?

  * ❌ No disengaging on ACC MAIN in OFF state.

</details>

<details><summary><h3>⚒ Installation</h3></summary>

 <details><summary>URL (Easy)</summary> 

  comma three
  ------
  
   Please refer to [Recommended Branches](#-recommended-branches) to find your preferred/supported branch, this guide will assume you want to install the latest release:

   1. [Factory reset/uninstall](https://github.com/commaai/openpilot/wiki/FAQ#how-can-i-reset-the-device) previous software if you already installed another fork.
   2. Select `Custom Software` when given the option upon reboot.
   3. Input the following URL: ```bit.ly/sp-test-c3```[^4] (note: `https://` is not requirement on the comma three)
   4. Complete the rest of the installation following the onscreen instructions.
   5. Pop into the `#installation-help` channel on Discord for further assistance.

  comma two
  ------

  Please refer to [Recommended Branches](#-recommended-branches) to find your preferred/supported branch, this will assume you want to install the latest release:

  1. Factory reset/uninstall previous software if you already installed another fork.
  2. Select `Custom Software` when given the option.

>   3. Input the following URL: ```https://smiskol.com/fork/sunnyhaibin/0.8.12-4-prod```

Reviewer: ```suggestion
  3. Input the following URL: ```https://smiskol.com/fork/sunnyhaibin/0.8.12-prod-full-_____```
```
0.8.12 based branch names are brand-specific. Maybe we include a link or a screenshot of the available `0.8.12-prod-full-____` branches here?

Author: Good point, could do it as a footnote to the either #-recommended-branches or a link / search query for the available branches, https://github.com/sunnyhaibin/sunnypilot/branches/all?query=0.8.12-prod-full ??

### Operational Enhancements
* **Fast Boot** - sunnypilot will fast boot by creating a Prebuilt file
* **Disable Onroad Uploads** - Disable uploads completely when onroad. Necessary to avoid high data usage when connected to Wi-Fi hotspot
* **Brightness Control (Global)** - Manually adjusts the global brightness of the screen
* **Driving Screen Off Timer** - Turns off the device screen or reduces brightness to protect the screen after car starts
* **Driving Screen Off Brightness (%)** - When using the Driving Screen Off feature, the brightness is reduced according to the automatic brightness ratio
* **Max Time Offroad** - Device is automatically turned off after a set time when the engine is turned off (off-road) after driving (on-road)

 </details>

<details><summary><h3>🚗 Driving Enhancements</h3></summary>
 
### Modified Assistive Driving Safety (MADS)
The goal of Modified Assistive Driving Safety (MADS) is to enhance the user driving experience with modified behaviors of driving assist engagements. This feature complies with comma.ai's safety rules as accurately as possible with the following changes:

> * sunnypilot Automatic Lane Centering and ACC/SCC can be engaged independently of each other

Reviewer: ```suggestion
* sunnypilot Automatic Lane Centering (ALC) and ACC/SCC can be engaged independently of each other
```
